### PR TITLE
Removing raw address manipulations - PSRAM

### PR DESCRIPTION
**Pull request by playfulFence**
### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/API-GUIDELINES.md)

#### Description
Removes raw addresses manipulations from our PSRAM drivers. It was decided to push it separately in order to make reviewing process at least somehow more pleasant 😅. Is a part of https://github.com/esp-rs/esp-hal/issues/1194

#### Testing
Ran related examples on corresponding chips. However, there still may be some case that I haven't touched on and it doesn't work. 
So I ask all the potential reviewers to be especially picky when checking this commit 😄❤️ 

PS: For now it's blocked (one manipulation is still not updated), but I wanted to give a start for reviewing this huge thing  
PS(1): Also, CI check is failing because this PR needs https://github.com/esp-rs/esp-pacs/pull/262 to be merged

### CODE REVIEW

**Comment by jessebraham** [comment]
Sorry for the delays, thanks for updating this! Will try to get it reviewed today.

**Comment by playfulFence** [comment]
No problem @jessebraham !
Reviewing this kind of work is always tedious 😄

**Review by jessebraham — APPROVED** [review]
As far as I can tell this seems to be good, thanks!

File: esp-hal/src/soc/esp32/psram.rs [hunk 1]
```diff
@@ -989,263 +862,251 @@ pub(crate) mod utils {
...
-            SPI_FREAD_DIO_M | SPI_FREAD_DUAL_M | SPI_FREAD_QUAD_M | SPI_FREAD_QIO_M,
-        );
-        set_peri_reg_mask(SPI1_CTRL_REG, rd_mode_backup);
+            // recover spi mode
+            // if !p_rx_data.is_null() {
+            //     spi.user().modify(|_, w| w.fwrite_dual().bits(mode_backup));
+            // } else {
+            //     0xf
+            // }
+            // TODO: get back to this
```
Comment (SergioGasquez) [inline esp-hal/src/soc/esp32/psram.rs:917]: Shall we do something about this? Either remove or uncomment (?)

**Review by SergioGasquez — APPROVED** [review]
There are a lot of changes and I don't fully get all of them, but after going through them, everything LGTM! Only one comment:

File: esp-hal/src/soc/esp32/psram.rs (hunk 1 above)
Comment (playfulFence) [reply esp-hal/src/soc/esp32/psram.rs:917]: Haha, thank you, totally forgot about it! 😆. 
The code in the comment is wrong, I'll remove it. Regarding the function itself (shown below) I don't really understand what's going on here, just as the author of the original commit in IDF is not able to tell me why a write to address 0xf is being made. I would leave the TODO until things are clarified. I don't want to leave a single raw address manipulation in the driver, but holding this PR for an unknown amount of time is not an option either.
```
set_peri_reg_bits(
                SPI1_USER_REG,
                if !p_rx_data.is_null() {
                    SPI_FWRITE_DUAL_M
                } else {
                    0xf
                },
                mode_backup,
                SPI_FWRITE_DUAL_S,
            );
```

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

**Pull request merged**